ui/widgets/controlgroup.js

```diff
@@ -117,15 +117,17 @@ return $.widget( "ui.controlgroup", {
 				.each( function() {
 					var element = $( this );
 					var instance = element[ widget ]( "instance" );
+					var instanceOptions = $.widget.extend( {}, options );
 
 					// If the button is the child of a spinner ignore it
 					if ( widget === "button" && element.parent( ".ui-spinner" ).length ) {
 						return;
 					}
 					if ( instance ) {
-						options.classes = that._resolveClassesValues( options.classes, instance );
+						instanceOptions.classes =
+							that._resolveClassesValues( instanceOptions.classes, instance );
 					}
-					element[ widget ]( options );
+					element[ widget ]( instanceOptions );
```

Comment:
Yes, but it's more than that. Even if I were to fix ```_resolveClassesValues()``` to not modify the value that is being passed in, you'd still need ```$.widget.extend()```, because the new object returned by ```_resolveClassesValues()``` would be stored inside ```options```, and ```options``` would be reused at the next iteration. So, it is ```options``` as well as ```options.classes``` that need to be cloned at every iteration.